solver/src/interactions/uniswap.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{
+    encoding::{self, INTERACTION_BASE_SIZE},
+    settlement::Interaction,
+};
+use anyhow::Result;
+use contracts::UniswapV2Router02;
+use primitive_types::{H160, U256};
+
+const UNISWAP_DATA_SIZE: usize = 260;
+const UNISWAP_TOTAL_SIZE: usize = INTERACTION_BASE_SIZE + UNISWAP_DATA_SIZE;
+
+#[derive(Debug)]
+pub struct UniswapInteraction {
+    pub uniswap_router_address: H160,
```

Comment:
How about using the actual instance here? Then we wouldn't need any dummy_web3. I think in the mid-term we also want to use the actual instance to check e.g. if we have already approved the token or we need to that as part of the interaction list.

Reply:
made this change